### ISSUE TITLE
Strip test selectors defined in component javascript

### ISSUE THREAD

**Issue by blimmer** [simplabs/ember-test-selectors#10]
This is a great idea! However, many times I find myself doing something like this:

``` javascript
export default Ember.Component.extend({
  attributeBindings: ['data-test-id'],
  'data-test-id': Ember.computed('model.id', function() {
     return `model-name-${id}-component`;
  }),
});
```

Just browsing the code, it looks like this wouldn't be stripped (since it's using a handlebars AST parser).

**Comment by marcoow** [comment]
We find it more convenient to have the data attributes in the templates as that also allows more flexibility since you can have data attributes on individual elements as well as opposed to only the component itself. Of course it would be nice to support this case as well though.

Would be great if you could submit a PR implementing that although I'm not sure how this would best be done.

**Comment by marcoow** [comment]
It would probably also make sense to by always (except for in production of course) add a `data-test-<component-name>` attribute to the component so you can always select the component itself, e.g. `data-test-blog-post`.

**Comment by marcoow** [comment]
I'm using this in a project now:

```js
// app/mixins/components/has-test-selector.js
import Ember from 'ember';
import ENV from '../../config/environment';

const { computed } = Ember;

let TestSelectorMixin;

if (ENV.environment !== 'production') {
  TestSelectorMixin = Ember.Mixin.create({
    attributeBindings: 'data-test-selector',

    'data-test-selector': computed(function() {
      return this.__proto__._debugContainerKey.replace(/^component:/, '');
    })
  });
} else {
  TestSelectorMixin = Ember.Mixin.create();
}

export default TestSelectorMixin;
```

I guess sth. like this could be added to `ember-test-selectors` as well.

**Comment by blimmer** [comment]
I plan to write a PR for this in the coming days (maybe at EmberConf!) if anyone is interested in helping.

**Comment by chrisdpeters** [comment]
@marcoow I like this idea.

Would it make more sense if it were something a tad more descriptive like `component`, rather than `selector` because the selector is being based on the component name? Just an idea/question; I really don't know if that's any better.

**Comment by marcoow** [comment]
@chrisdpeters: yeah, sounds like a good idea.

**Comment by bcardarella** [comment]
@marcoow any update on this?

**Comment by marcoow** [comment]
I'm hoping someone will find some time to do this soon. It would definitely be a great addition. Someone on our team might actually be able to pick this up in the next weeks but we've all been pretty busy lately.

**Comment by marcoow** [comment]
This has been fixed via #45
